### PR TITLE
cooldown for location

### LINKED ISSUE #31: add cooldown to location command

### PR DESCRIPTION
**Pull request by Gobsmack90**
adds a 30 minute cooldown on the location function.

fixes #31

### CODE REVIEW

File: app/commands/location.js [hunk 1]
```diff
@@ -49,16 +49,22 @@ module.exports = Base => class extends Base {
...
+            'Set location ', { text: newLocation.name, color: 'aqua' }, ` to the position of [${newLocation.pos.join(', ')}]`,
         ]);
     }
 
     async checkLocationAndTeleport(playerName, locationName) {
         let location = await this.readLocation(locationName);
         if (!location) return false;
 
+        //cooldown check goes here
+        if (this.cooldownCheck('location', playerName) == true) return true;
```
Comment (thorbaert) [inline app/commands/location.js:61]: Wouldn't it make sense to return false like is done on like 58

File: app/lib/cooldown.js [hunk 2]
```diff
@@ -7,7 +7,8 @@ module.exports = Base => class extends Base {
         //object containing the cooldown times for the commands. new commands would need a new property added here.
         this.coodownTimes = {
             'home': 1000 * 60 * 15,
-            'warp': 1000 * 60 * 15
+            'warp': 1000 * 60 * 15,
+            'location': 1000 * 60 * 30
```
Comment (thorbaert) [inline app/lib/cooldown.js:11]: If we are going to have all locations on a single cooldown  wouldn't it make sense to be a bit more explicit to the end user what is happening if it doesn't work

I think the message right now would be something like `command location is on cooldown` which doesn't make sense for `!cove` ya know? 

Maybe pass the location name as a optional parameter?

**Review by thorbaert — CHANGES_REQUESTED** [review]

File: app/lib/cooldown.js (hunk 2 above)
Comment (thorbaert) [reply app/lib/cooldown.js:11]: It might even make sense to have a `optionalRawMessage` that is sent instead of the generic message

Comment (Gobsmack90) [reply app/lib/cooldown.js:11]: i can probably figure that out. I'll get started on that and counting down on the cooldown message. for now it's functional. You can decide if you want to update with what we have or not

Comment (thorbaert) [reply app/lib/cooldown.js:11]: Unless the code we are adding is fixing a bug it is not a good practice to add only functional especially if wee know we will have to ad a new issue

**Pull request closed without merging**